x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.test.tsx

```diff
@@ -0,0 +1,338 @@
+    ).toHaveValue(String(rowHeight.initial));
+
+    fireEvent.change(
+      screen.getAllByTestId('unifiedDataTableRowHeightSettings_lineCountNumber')[0],
+      {
+        target: { value: String(rowHeight.new) },
+      }
+    );
+
+    await waitFor(() => {
+      expect(customMockStore.getState().timeline.timelineById[TimelineId.test]?.rowHeight).toEqual(
+        rowHeight.new
+      );
+    });
+  });
+
+  describe('details flyout', () => {
+    it('should show defails flyout when clicked on expand event', async () => {
+      render(<TestComponent showExpandedDetails={true} />);
+
+      expect(await screen.findByTestId('discoverDocTable')).toBeVisible();
+
+      fireEvent.click(screen.getAllByTestId('docTableExpandToggleColumn')[0]);
+
+      await waitFor(() => {
+        expect(screen.getByTestId('timeline:details-panel:flyout')).toBeVisible();
+      });
+    });
+
+    it('should show details flyout when expandedDetails state is set', async () => {
+      const customMockStore = createMockStore();
+      const mockExpandedDetail: ExpandedDetailTimeline = {
+        query: {
+          params: {
+            eventId: 'some_id',
+            indexName: 'security-*',
+          },
+          panelView: 'eventDetail',
+        },
+      };
+      customMockStore.dispatch(
+        timelineActions.toggleDetailPanel({
+          id: TimelineId.test,
+          tabType: TimelineTabs.query,
+          ...mockExpandedDetail.query,
+        })
+      );
+
+      render(
+        <TestComponent
+          store={customMockStore}
+          showExpandedDetails={true}
+          expandedDetail={mockExpandedDetail}
+        />
+      );
+
+      await waitFor(() => {
+        expect(screen.getByTestId('timeline:details-panel:flyout')).toBeVisible();
+      });
+    });
+    it('should close details flyout when close icon is clicked', async () => {
+      const customMockStore = createMockStore();
+      const mockExpandedDetail: ExpandedDetailTimeline = {
+        query: {
+          params: {
+            eventId: 'some_id',
+            indexName: 'security-*',
+          },
+          panelView: 'eventDetail',
+        },
+      };
+
+      customMockStore.dispatch(
+        timelineActions.toggleDetailPanel({
+          id: TimelineId.test,
+          tabType: TimelineTabs.query,
+          ...mockExpandedDetail.query,
+        })
+      );
+
+      render(
+        <TestComponent
+          store={customMockStore}
+          showExpandedDetails={true}
+          expandedDetail={mockExpandedDetail}
+        />
+      );
+
+      await waitFor(() => {
+        expect(screen.getByTestId('euiFlyoutCloseButton')).toBeVisible();
+      });
+
+      fireEvent.click(screen.getByTestId('euiFlyoutCloseButton'));
+      expect(onEventClosedMock).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('pagination', () => {
+    // change the number of items per page
+    it('should change the number of items per page', async () => {
```

Comment:
super nit: Could we validate the default prior to the change? I'm pretty sure the default number of items per page is `10`, but if for whatever reason this switches to `25` and the select breaks this wouldn't catch it.

Reply:
Actually, here it is not 10. It is based on the prop being passed to this component. 

It is being checked in query_tab's test [here](https://github.com/elastic/kibana/pull/176064/files#diff-2bb9897670b87c7f3f9422878327fe5960b3963c44c112decfc747aa124a572aR193-R201) and default value is 5 in tests 😄 . Ideally we should have kept it same as live env.. but currently, this is the case.

Reply:
Although, I still have added a statement to verify the initial value of `items per page`.

Reply:
Oh well maybe before it was ever 10 the default was 5...who knows 🤷🏾‍♂️ , but thanks for verifying :)